### PR TITLE
Embed Improvements

### PR DESCRIPTION
**Pull request by hariombalhara**
- [Embed UI Changes]((https://www.loom.com/share/7313f7db9e97442f8535423d295acd9a) to make it look better in popup or inline.
- Text Colors UI Configuration Support(lightest, lighter, light, median, dark, darker). See how you can get embed working with transparent background by controlling the text colors. [Transparent Background Demo](https://www.loom.com/share/65afabacfc8743cb8f8501b7b06fd6f1)
- Custom Branding Support in API.
- Missing things that I discovered while documentation. Mostly feature parity across inline/popup and Vanilla JS and React flavours.
- [ Loader in Popup ](https://www.loom.com/share/7313f7db9e97442f8535423d295acd9a)

Automation
- Tests now run on Safari and Firefox as well but not in CI yet(actually caught a bug in Safari due to those :))

Documentation
- docs.cal.com/embed/integration

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Booking Link shouldn't have any UI change.
- [ ] Embed Booking link should be changed as shown in the demo above.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by PeerRich — APPROVED** [review]
great idea to add a `isEmbed` hook!

File: apps/web/pages/404.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { ChevronRightIcon } from "@heroicons/react/solid";
 import { GetStaticPropsContext } from "next";
 import Link from "next/link";
 import { useRouter } from "next/router";
+import Script from "next/script";
```
Comment (zomars) [inline apps/web/pages/404.tsx:6]:
```suggestion
```

File: apps/web/pages/_app.tsx [hunk 2]
```diff
@@ -22,13 +23,21 @@ import "../styles/fonts.css";
...
 function MyApp(props: AppProps) {
-  const { Component, pageProps, err } = props;
+  const { Component, pageProps, err, router } = props;
+  let pageStatus = "200";
+  if (router.pathname === "/404") {
+    pageStatus = "404";
+  } else if (router.pathname === "/500") {
+    pageStatus = "500";
+  }
+  console.log(router.pathname, pageStatus);
```
Comment (zomars) [inline apps/web/pages/_app.tsx:33]:
```suggestion
```

**Review by zomars — COMMENTED** [review]
Minor nitpicks but overall LGTM.

**Pull request merged**